v2/tools/asoctl/internal/client/client.go

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package client
+
+import (
+	"github.com/pkg/errors"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+func NewClient() (*v1.ApiextensionsV1Client, error) {
```

Comment:
I'm not totally sold that we actually need this helper, since:
1. It seems a pretty thin wrapper for the already existing `v1.NewForConfig(cfg)`
2. If we had multiple different types of clients we needed to construct, it probably would make sense to just call `config.GetConfigOrDie` once and then pass it to each of the `v1.NewForConfig` equivalents. Otherwise we're doing excess file reads. If we made that change, we're now _exactly_ wrapping `v1.NewForConfig` which isn't super useful?

So IMO just remove this and do it in the `crd_clean.go` file directly.